src/components/c-google-maps.vue

```diff
@@ -0,0 +1,569 @@
+      elementType: 'all',
+      stylers: [{ saturation: '-100' }],
+    },
+    {
+      featureType: 'administrative.province',
+      elementType: 'all',
+      stylers: [{ visibility: 'off' }],
+    },
+    {
+      featureType: 'landscape',
+      elementType: 'all',
+      stylers: [
+        {
+          saturation: -100,
+        },
+        {
+          lightness: 65,
+        },
+        {
+          visibility: 'on',
+        },
+      ],
+    },
+    {
+      featureType: 'landscape.man_made',
+      elementType: 'all',
+      stylers: [
+        {
+          lightness: -10,
+        },
+      ],
+    },
+    {
+      featureType: 'poi',
+      elementType: 'all',
+      stylers: [
+        {
+          saturation: -100,
+        },
+        {
+          lightness: '50',
+        },
+        {
+          visibility: 'simplified',
+        },
+      ],
+    },
+    {
+      featureType: 'road',
+      elementType: 'all',
+      stylers: [
+        {
+          saturation: '-100',
+        },
+      ],
+    },
+    {
+      featureType: 'road.highway',
+      elementType: 'all',
+      stylers: [
+        {
+          visibility: 'simplified',
+        },
+      ],
+    },
+    {
+      featureType: 'road.arterial',
+      elementType: 'all',
+      stylers: [
+        {
+          lightness: '30',
+        },
+      ],
+    },
+    {
+      featureType: 'road.local',
+      elementType: 'all',
+      stylers: [
+        {
+          lightness: '40',
+        },
+      ],
+    },
+    {
+      featureType: 'transit',
+      elementType: 'all',
+      stylers: [
+        {
+          saturation: -100,
+        },
+        {
+          visibility: 'simplified',
+        },
+      ],
+    },
+    {
+      featureType: 'water',
+      elementType: 'geometry',
+      stylers: [
+        {
+          hue: '#ffff00',
+        },
+        {
+          lightness: -25,
+        },
+        {
+          saturation: -97,
+        },
+      ],
+    },
+    {
+      featureType: 'water',
+      elementType: 'labels',
+      stylers: [
+        {
+          lightness: -25,
+        },
+        {
+          saturation: -100,
+        },
+      ],
+    },
+  ];
+
+  /**
+   * Set isMapsAPILoaded to true, call the callback, then remove all callbacks in callbackStack array.
+   */
+  window.cGoogleMapsInitMap = () => {
+    isMapsAPILoaded = true;
+    callbackStack.forEach(callback => callback());
+    callbackStack.length = 0;
+  };
+
+  /**
+   * This function checks whether the Map API has already been loaded once.
+   */
+  function loadMapsAPI(callback: TCGoogleMapsCallback): void {
+    const useSessionStore = sessionStore();
+    const apiKey = useSessionStore.googleMapsApiKey;
+
+    if (!apiKey) {
+      throw new Error('No Google Maps API key provided.');
+    }
+
+    if (isMapsAPILoaded) {
+      callback();
+    } else {
+      const url = `https://maps.googleapis.com/maps/api/js?key=${apiKey}&libraries=places&callback=cGoogleMapsInitMap`;
```

Comment:
todo: please use a variable for `cGoogleMapsInitMap` and also use it to define the callback function on `window`.